src/Microsoft.TestPlatform.Client/Execution/TestRunRequest.cs

```diff
@@ -662,10 +662,14 @@ public int LaunchProcessWithDebuggerAttached(TestProcessStartInfo testProcessSta
     }
 
     /// <inheritdoc />
-    public bool AttachDebuggerToProcess(int pid)
+    public bool AttachDebuggerToProcess(AttachDebuggerInfo attachDebuggerInfo)
```

Comment:
Without that new interface, I would have to keep this method that just takes an int, when I would implement ITestRunEventsHandler3. And then make sure that I correctly update all the subsequent consumers, because they take ITestRunEventsHandler2 or ITestRunEventsHandler, and my ITestRunEventsHandler3 implements that. So there would be no compilation error, and I would most likely lose the targetFramework data somewhere along the lines.